addons/web_tip/static/src/js/tip.js

```diff
@@ -135,187 +107,231 @@ var Tip = Class.extend({
+        });
+        this.$overlay.on('click', function($ev) {
+            self.end_tip();
+        });
+        $(document).on('keyup.web_tip', function($ev) {
+            if ($ev.which === 27) { // esc
+                self.end_tip();
+            }
+        });
+
+        // resize
+        bus.on('resize', this, function() {
+            self._set_tip_reposition();
+        });
+
+        bus.on('please_reposition_tip', this, function () {
+            self._set_tip_reposition();
+        });
     },
 
     scroll_to_tip: function(){
-        var self = this;
-        var scroll = _.find(self.$element.parentsUntil('body'), function(el) {
+        var scroll = _.find(this.$element.parentsUntil('body'), function(el) {
             var overflow = $(el).css('overflow-y');
             return (overflow === 'auto' || overflow === 'scroll');
         });
         if (scroll) {
-            $(scroll).scrollTo(self.$element);
+            $(scroll).scrollTo(this.$element);
         }
     },
 
-    end_tip: function(tip) {
-        var self = this;
+    end_tip: function() {
         var Tips = new Model('web.tip');
-        self.$element.popover('destroy');
-        self.$overlay.remove();
-        self.$helper.remove();
-        self.$element.removeClass('oe_tip_show_element');
+
+        this.$element.popover('destroy');
+        this.$element.removeClass('oe_tip_show_element');
+        this.$breathing.remove();
+        this.$helper.remove();
+        this.$overlay.remove();
+        this.$cross.remove();
+
         _.each($('.oe_tip_fix_parent'), function(el) {
             $(el).removeClass('oe_tip_fix_parent');
         });
         $(document).off('keyup.web_tip');
-        Tips.call('consume', [tip.id], {});
-        tip.is_consumed = true;
+
+        Tips.call('consume', [this.tip.id], {});
+        this.tip.is_consumed = true;
     },
 
-    reposition: function() {
-        var self = this;
-        if (self.tip_mutex.def.state() === 'pending') {
-            self.scroll_to_tip();
-            self._set_helper_position();
-            self.$element.popover('show');
-        }
+    _set_breathing_position: function () {
+        this.$breathing.position({ my: "center", at: "center", of: this.$element });
     },
 
-    _set_helper_position : function() {
+    _set_helper_position: function() {
         var offset = this.$element.offset();
         var _top = offset.top - 5;
         var _left = offset.left - 5;
         var _width = this.$element.outerWidth() + 10;
         var _height = this.$element.outerHeight() + 10;
-        this.$helper.offset({top: _top , left: _left});
+
+        this.$helper.offset({top: _top, left: _left});
         this.$helper.width(_width);
         this.$helper.height(_height);
-    }
+    },
+
+    _set_tip_reposition: function() {
+        if (!this.tip.is_consumed) {
+            this.$element.popover('show');
+            this._set_helper_position();
+            this.scroll_to_tip();
```

Comment:
I don't understand. Why do we need to scroll to the tip when repositionning it ?